src/views/ManageList.jsx

```diff
@@ -1,7 +1,95 @@
+import { useState } from 'react';
+import { addItem } from '../api';
+
 export function ManageList() {
+	const [formData, setFormData] = useState({
+		name: '',
+		frequency: '',
+	});
+
+	// const [formSubmitted, setFormSubmitted] = useState(false);
+
+	function handleChange(e) {
+		e.preventDefault();
+		setFormData((prev) => ({
+			...prev,
+			[e.target.name]: e.target.value,
+		}));
+	}
+
+	function handleSubmit(e) {
+		e.preventDefault();
+		if (formData.name.trim() === '') {
+			window.alert('Item name cannot be empty.');
+			return;
+		}
+
+		console.log('here is form submission', formData);
+
+		const listPath = localStorage.getItem('tcl-shopping-list-path');
+
+		if (!listPath) {
+			window.alert('List is not existed.');
+			return;
+		}
+
+		addItem(listPath, {
+			itemName: formData.name,
+			daysUntilNextPurchase: parseInt(formData.frequency),
+		})
+			.then(() => {
+				window.alert(`${formData.name} has been added to your list.`);
+				setFormData({
+					name: '',
+					frequency: '',
+				});
+			})
+			.catch((error) => {
+				window.alert(`${formData.name} failed to add to your list.`);
+				console.error('Error:', error);
+			});
+	}
```

Comment:
<!-- Highlight something to the reader.  -->
**note (non-blocking):** This does read very cleanly! I like that you considered that in your usage here. And it does work well here, but it may be worthwhile to know that it's not an exact swap for using `async/await` see [here](https://dev.to/masteringjs/using-then-vs-async-await-in-javascript-2pma) for a document on it!